src/execution_engines/node_js/index.ts

```diff
@@ -5,25 +5,24 @@ import { Page } from 'puppeteer'
 
 let contracts: {
   [contractAddress: string]: {
-    executable: string
     page: Page
   }
 } = {}
 
 const NodeEngine: ExecutionEngine = {
   load: async ({ contractAddress, executable }) => {
-    const contractPage = await loadPage()
-    contracts[contractAddress] = { executable, page: contractPage }
+    const contractPage = await loadPage(executable)
+    contracts[contractAddress] = { page: contractPage }
     return true
   },
   execute: async ({ contractAddress, method, methodArgs }) => {
-    if (!(methodArgs instanceof Object)) {
+    if (methodArgs && !(methodArgs instanceof Object)) {
       throw new BadArgument(typeof methodArgs)
     }
 
     const contract = contracts[contractAddress]
     if (!contract) throw new ContractNotLoaded()
-    const data = await executeInBrowser(contract.page, contract.executable, method, methodArgs)
+    const data = await executeInBrowser(contract.page, method, methodArgs)
```

Comment:
```suggestion
    const data = await executeInBrowser(contract, method, methodArgs)
```
Keeps the implementation (of pages) out of the way